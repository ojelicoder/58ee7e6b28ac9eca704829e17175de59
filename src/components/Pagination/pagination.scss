@mixin itemStyle{
    cursor: pointer;
    margin: 5px;
}

.pagination{
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 10px 25px 25px;

    &__page{
        @include itemStyle();
        padding: 2px 7px;
        font-weight: 700;
        font-size: 15px;

        &.active{
            border: 1px solid
        }
    }
    &__next{
        @include itemStyle();
    }
    &__prev{
        @include itemStyle();
    }
}