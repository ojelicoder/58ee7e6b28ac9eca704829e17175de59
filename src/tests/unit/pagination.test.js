import React from 'react';
import reactDom from 'react-dom';
import Pagination from '../../components/Pagination';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

const paginationProp = {
    totalPage : 3,
    page: 2,
    setPage: 2
};

describe('Pagination', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        reactDom.render(<Pagination props={paginationProp} />, div);
    })

    it('matches snapshot', () => {
        const tree = renderer.create(<Pagination paginationProp={{totalPage: 10}} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})