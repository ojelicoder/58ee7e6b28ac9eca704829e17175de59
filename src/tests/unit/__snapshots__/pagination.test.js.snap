// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<div
  className="pagination"
>
  <div
    className="pagination__prev"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-left "
      data-icon="chevron-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224 480c-8.188 0-16.38-3.125-22.62-9.375l-192-192c-12.5-12.5-12.5-32.75 0-45.25l192-192c12.5-12.5 32.75-12.5 45.25 0s12.5 32.75 0 45.25L77.25 256l169.4 169.4c12.5 12.5 12.5 32.75 0 45.25C240.4 476.9 232.2 480 224 480z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
